### PR TITLE
clean up all white noise

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <764524258@qq.com>

* [x] Change is not relevant to the end user.

## Changes

Remove all white noise.

### CODE REVIEW

**Comment by daixiang0** [comment]
If possible, i want to add a tool for this when any docs updated.

**Comment by bwplotka** [comment]
Yes, please, can we add some linting for his? Maybe in this PR? (:

**Comment by daixiang0** [comment]
i would do it, where is the best to add this? In circleci config file?

**Comment by bwplotka** [comment]
Yes

**Comment by daixiang0** [comment]
@bwplotka mind have a look?

File: scripts/cleanup-white-noise.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+sed -i 's/[ \t]*$//' "$@"
```
Comment (bwplotka) [inline scripts/cleanup-white-noise.sh]: Nice! I think we might want to have it in `Makefile` as well. We can hook it on the CI in later PR.

Comment (daixiang0) [reply scripts/cleanup-white-noise.sh]: format `*.go`?

**Comment by daixiang0** [comment]
ping @bwplotka

**Comment by daixiang0** [comment]
mind review again? @bwplotka

**Comment by daixiang0** [comment]
mind review again? @bwplotka

**Comment by daixiang0** [comment]
I want to do those changes to make target more clear:

1. `docs` only generate docs
2. `format` only call `go format -w`
3. `docs-check`  would call `docs` target first then clean white noise
4. `format-check` would call `format` target first then clean white noise

Wait for your feedback @bwplotka

**Comment by bwplotka** [comment]
That's not what I proposed (: 

`*-check` would always VERIFY and return non 0 signal code if something is not done. Actually for `format` we have `lint` already so let's use that (: 
`format` and `docs` would `fix` things in you local repo for you to commit it and include in the PR (: 


So:
* `docs`: generate docs + **clean whitespace in docs**
* `format`: format + **clean whitespace in Go code**
* `docs-check` if docs not generated and **if any whitespace not cleaned** return error
* `lint` if Go file not formatted / lint found issues/ misspell or **whitespeaces not cleaned** return error (:  

In bold font things that are would like to change in this PR (: Does it make sense? (:

**Comment by daixiang0** [comment]
@bwplotka thanks for your patience, please review again.

**Comment by daixiang0** [comment]
@bwplotka thanks for your patience, please review again.

**Pull request merged**